renku/service/views/cache.py

```diff
@@ -0,0 +1,335 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2019 - Swiss Data Science Center (SDSC)
+# A partnership between École Polytechnique Fédérale de Lausanne (EPFL) and
+# Eidgenössische Technische Hochschule Zürich (ETHZ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Renku service cache views."""
+import os
+import shutil
+import time
+from pathlib import Path
+from urllib.parse import urlparse
+
+import jwt
+from flask import jsonify, request
+from git import Repo
+from werkzeug.utils import secure_filename
+
+from renku.service.config import CACHE_PROJECTS_PATH, CACHE_UPLOADS_PATH, \
+    INTERNAL_FAILURE_ERROR_CODE, INVALID_PARAMS_ERROR_CODE, JWT_ALGORITHM, \
+    JWT_KEY
+from renku.service.views import allowed_file, requires_identity, \
+    validate_git_url
+
+
+@requires_identity
+def list_uploaded_files_view(user):
+    """
+    List uploaded files ready to be added to projects.
+
+    ---
+    get:
+        tags:
+        - "cache"
+
+        summary: List of uploaded files.
+        description: List of cached files.
+
+        parameters:
+            - in: header
+              name: Authorization
+              schema:
+                type: string
+              required: true
+              description: User identification.
+
+        responses:
+            result:
+                description: |
+                  Contains `files` key which contains
+                  list of all files for a given user.
+
+            error:
+                description: |
+                  Contains `code` and `message` describing the error.
+
+    """
+    if request.method == 'GET':
+        files = []
+        for file_name in os.listdir(CACHE_UPLOADS_PATH):
```

Comment:
use `os.walk` instead

Reply:
1) No. See below.

2) Let's talk about this, there is a tradeoff.